rs/ic_os/config/src/update_config.rs

```diff
@@ -0,0 +1,574 @@
+    let filebeat_conf_path = config_dir.join("filebeat.conf");
+    let conf_map = match read_conf_file(&filebeat_conf_path) {
+        Ok(map) => map,
+        Err(_) => {
+            // Set default values if filebeat.conf doesn't exist
+            return Ok(Logging {
+                elasticsearch_hosts: "elasticsearch-node-0.mercury.dfinity.systems:443 \
+                                       elasticsearch-node-1.mercury.dfinity.systems:443 \
+                                       elasticsearch-node-2.mercury.dfinity.systems:443 \
+                                       elasticsearch-node-3.mercury.dfinity.systems:443"
+                    .to_string(),
+                elasticsearch_tags: None,
+            });
+        }
+    };
+
+    let elasticsearch_hosts = conf_map
+        .get("elasticsearch_hosts")
+        .cloned()
+        .unwrap_or_default();
+
+    let elasticsearch_tags = conf_map.get("elasticsearch_tags").cloned();
+
+    Ok(Logging {
+        elasticsearch_hosts,
+        elasticsearch_tags,
+    })
+}
+
+fn read_nns_conf(config_dir: &Path) -> Result<Vec<Url>> {
+    let nns_conf_path = config_dir.join("nns.conf");
+    let conf_map = match read_conf_file(&nns_conf_path) {
+        Ok(map) => map,
+        Err(_) => {
+            // Set default values if nns.conf doesn't exist
+            let default_urls = vec![
+                Url::parse("https://icp-api.io")?,
+                Url::parse("https://icp0.io")?,
+                Url::parse("https://ic0.app")?,
+            ];
+            return Ok(default_urls);
+        }
+    };
+
+    let nns_url_str = conf_map.get("nns_url").cloned().unwrap_or_default();
+
+    let mut nns_urls = Vec::new();
+    for s in nns_url_str.split(',') {
+        let s = s.trim();
+        match Url::parse(s) {
+            Ok(url) => nns_urls.push(url),
+            Err(e) => {
+                println!("Invalid URL '{}': {}", s, e);
+            }
+        }
+    }
+
+    Ok(nns_urls)
+}
+
+fn derive_mgmt_mac_from_hostname(hostname: Option<&str>) -> Result<FormattedMacAddress> {
+    if let Some(hostname) = hostname {
+        if let Some(unformatted_mac) = hostname.strip_prefix("guest-") {
+            // Insert colons into mac_str to format it as a MAC address
+            if unformatted_mac.len() != 12 {
+                return Err(anyhow::anyhow!(
+                    "Invalid MAC address length in hostname: {}",
+                    hostname
+                ));
+            }
+            let formatted_mac = unformatted_mac
+                .chars()
+                .collect::<Vec<_>>()
+                .chunks(2)
+                .map(|chunk| chunk.iter().collect::<String>())
+                .collect::<Vec<_>>()
+                .join(":");
+            let formatted_mac = FormattedMacAddress::try_from(formatted_mac.as_str())
+                .with_context(|| format!("Failed to parse mgmt_mac from hostname: {}", hostname))?;
```

Comment:
We still have these two MAC libraries floating around. It may be worth it to start to consolidate on one of them. `deterministic_ips` is a little more general, and supports [this parsing](https://github.com/dfinity/ic/blob/fad7e160bf27ae74d0d614c411d957947ade1d90/rs/ic_os/dev_test_tools/deterministic_ips/src/lib.rs#L77) with:

```
let mac: HwAddr = "70B5E8E825DE".parse()?;
```

Reply:
Currently, the new config types holds mac_address::FormattedMacAddress, so that's why I'm using FormattedMacAddress here

Do you think we should have the config types hold a HwAddr type instead? I'm happy delete mac_address and consolidate everything on deterministic_ips if you think that's best

Reply:
Maybe we can file a small ticket to converge these. I'd be interested in picking it up before we fully roll out config or get into the network code, but we shouldn't let it get in the way of this work for now.

Reply:
If it's going to effect the config types, then I'd rather consolidate them now before merging this. I'll move this back to draft and push on that. I don't think it will be too much work.